ailiuwSoloProject/ailiuWU_CUI_Project/Market.js

```diff
@@ -0,0 +1,246 @@
+const instructions = () => {
+	//print instructions
+	console.log('From here, you could:')
+	console.log('')
+	console.log('1.Add grocercy product to the cart')
+	console.log('You could simply use natural language to add products with the number you want.')
+	console.log('Remember to include the amount and unit for products.')
+	console.log('For example: you could say \'Add three bananas to the cart\'')
+	console.log('Or you could say \'Can I have 1 ounce of shampoo\'')
+	console.log('')
+	console.log('2.View the products in certain apartment')
+	console.log('You could view products list in a department by natural language')
+	console.log('For example: you could say \'I want some breakfast\'')
+	console.log('')
+	console.log('3.View the products in your cart')
+	console.log('you could view the products in your cart by including \'view cart\' in your request')
+	console.log('')
+	console.log('4.Check out')
+	console.log('you could check out by including \'check out\' in your request')
+	console.log('')
+	console.log('5.Ask for instructions')
+	console.log('Any time you feel comfused. You could simply type \'help\'. or \'instructions\'')
+	console.log('')
+}
+
+/*ask for customers intention*/
+const askForIntention = () => {
+  return dialog.question('What would you like to do next?')
+}
+
+/*handle customer intents*/
+const customerIntents=(answer)=>{
+	return new Promise(function (resolve, reject) {
+			let request = AIBot.textRequest(answer, {
+				sessionId: 'grocerypurchase-206e2'
+			})
+					 
+			request.on('response', function(response) {
+				//console.log(response)
+				let intent=response.result.metadata.intentName				
+				//check intent
+				switch (intent) {
+					case 'Discover Apartments':
+						let department=response.result.parameters.Department;
+						//print products in the department
+						console.log('Here are what we find at the '+department+' department:')
+						let productJson=jsonContent["entities"]
+						console.log(sprintf("%-20s%-10s%-10s", 'Product','Unit','UnitPrice'))						
+						for(let i=0;i<productJson.length;i++){
+							if(productJson[i]["department"]==department){
+								console.log(sprintf("%-20s%-10s%-10f",productJson[i]["value"],productJson[i]["unit"],productJson[i]["price"]))
+							}
+						}
+						console.log('')
+						return askForIntention().then(customerIntents)
+					case 'Add Product':						
+						//store products into the cart
+						let product=response.result.parameters.Product
+						let productNum=parseInt(response.result.parameters.number)
+						let productUnit=''
+						let productPrice=0.0
+						//check if all value are get
+						if((typeof product==='string')&&(productNum>0)){
+							console.log(response.result.fulfillment.speech)
+							console.log('')
+						}
+						else{
+							console.log('Sorry. We can\'t identify what you want')
+							return dialog.question('Could you please re-enter what you want to add to cart')
+								.then(customerIntents)
```

Comment:
There's a good opportunity just to ask the user how much they want, when a product is present but a quantity is missing. The dialog would flow like this:

```Shell
> add bananas to my cart
How many bananas would you like?
> 6
6 bananas were added to your cart
```